// @ankitdev/my-card/lib/components/Card2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireDefault(require("react"));
require("./Card2.css");
var _framerMotion = require("framer-motion");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function Card2(_ref) {
  var _ref$title = _ref.title,
    title = _ref$title === void 0 ? "Nature's Light" : _ref$title,
    _ref$spots = _ref.spots,
    spots = _ref$spots === void 0 ? "450 SPOTS" : _ref$spots,
    imageUrl = _ref.imageUrl,
    Icon = _ref.icon,
    _ref$gradient = _ref.gradient,
    gradient = _ref$gradient === void 0 ? 'linear-gradient(135deg, #f36265, #961276)' : _ref$gradient,
    _ref$size = _ref.size,
    size = _ref$size === void 0 ? {
      width: '340px',
      height: '140px'
    } : _ref$size;
  return /*#__PURE__*/_react["default"].createElement(_framerMotion.motion.div, {
    className: "stylish-card",
    style: {
      backgroundImage: "url(".concat(imageUrl, ")"),
      width: size.width,
      height: size.height
    },
    initial: {
      opacity: 0,
      y: 30
    },
    animate: {
      opacity: 1,
      y: 0
    },
    whileHover: {
      scale: 1.05
    },
    transition: {
      duration: 0.5,
      ease: 'easeOut'
    }
  }, /*#__PURE__*/_react["default"].createElement(_framerMotion.motion.div, {
    className: "overlay",
    style: {
      background: gradient
    },
    animate: {
      backgroundPosition: ['0% 50%', '100% 50%', '0% 50%']
    },
    transition: {
      duration: 10,
      repeat: Infinity,
      ease: 'linear'
    }
  }), /*#__PURE__*/_react["default"].createElement("div", {
    className: "card-info"
  }, Icon && /*#__PURE__*/_react["default"].createElement(Icon, {
    className: "card-icon"
  }), /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement("h3", {
    className: "card-title"
  }, title), /*#__PURE__*/_react["default"].createElement("p", {
    className: "card-subtitle"
  }, spots))));
}
var _default = exports["default"] = Card2;
